error TS4058 error about  CarouselConfig

**Describe the bug**
A clear and concise description of what the bug is.

Using Vue3 and the latest 0.3.4 version of the vue3-carousel i import the component and the slider and then simply pass props to it.

In dev mode all works as expected, but when building with ts check it errors out with the message

```bash
src/components/ui/CarouselComp.vue:320:10 - error TS4058: Return type of exported function has or is using name 'CarouselConfig' from external module "C:/Code/vue-modules/node_modules/vue3-carousel/dist/carousel" but cannot be named.
```

**To Reproduce**
Steps to reproduce the behavior:

import the carousel in Vue3 component

```js
import { Carousel as SliderComp, Slide as SlideComp } from 'vue3-carousel';
import 'vue3-carousel/dist/carousel.css';
```

the template contains the 

```vue
  <SliderComp
      v-model="currentSlide"
      :snap-align="snapAlign"
      :items-to-show="itemsToShow"
      :breakpoints="breakpoints"
      :wrap-around="wrapAround"
      ref="carouselRef"
      :id="uniqueId('carousel-')"
      @slide-end="updateCurrentIndex"
    >
      <template #slides>
        <SlideComp
          v-for="(slider, ind) in results"
          :key="ind"
        >
          <div class="flex h-full flex-col items-stretch justify-start gap-2 pr-2 w-full">
            <div
              class="wm-bg--base wm-text--base grow text-center font-normal shadow hover:shadow-lg"
            >
              <slot
                :slider="slider"
                name="slider"
              />
            </div>
          </div>
        </SlideComp>
      </template>
    </SliderComp>
```

the slider slot is names to that i can easily access the props when using the vue component



**Expected behavior**
 
i would expect the build and ts-check to actually work

**Screenshots**


![image](https://github.com/user-attachments/assets/f2b37f41-5c2d-4443-998f-3b0dfcd2431a)

**Desktop (please complete the following information):**

- OS: windows
- Browser Brave
- Version Version 1.70.123 Chromium: 129.0.6668.89 

**Smartphone (please complete the following information):**

n/a

**Additional context**

build script
```json
 "build": "run-p type-check \"build-only {@}\" --",
```

typescript  related
```json
 "vue-tsc": "2.1.6",
  "typescript": "~5.6.3",
 "vite-plugin-dts": "4.2.3",
  "vite-tsconfig-paths": "^5.0.1",
  "npm-run-all2": "6.2.3", 
  "@vue/tsconfig": "0.5.1",
  "@vue/eslint-config-typescript": "14.0.0",
  "vue": "3.5.11",

--- a/src/components/Carousel.ts
+++ b/src/components/Carousel.ts
@@ -18,7 +18,7 @@ import {
 
 import { DEFAULT_CONFIG } from '@/partials/defaults'
 import { carouselProps } from '@/partials/props'
-import { CarouselConfig, CarouselNav } from '@/types'
+import { CarouselConfig, CarouselExposed, CarouselNav } from '@/types'
 import {
   debounce,
   throttle,
@@ -36,6 +36,15 @@ import SlideComponent from './Slide'
 export default defineComponent({
   name: 'Carousel',
   props: carouselProps,
+  emits: [
+    'init',
+    'drag',
+    'slide-start',
+    'loop',
+    'update:modelValue',
+    'slide-end',
+    'before-init',
+  ],
   setup(props: CarouselConfig, { slots, emit, expose }: SetupContext) {
     const root: Ref<Element | null> = ref(null)
     const viewport: Ref<Element | null> = ref(null)
@@ -412,15 +421,13 @@ export default defineComponent({
       config,
       slidesCount,
       slideSize,
-      next,
-      prev,
-      slideTo,
       currentSlide: currentSlideIndex,
       maxSlide: maxSlideIndex,
       minSlide: minSlideIndex,
       middleSlide: middleSlideIndex,
     }
-    expose({
+
+    expose<CarouselExposed>({
       updateBreakpointsConfig,
       updateSlidesData,
       updateSlideSize,

--- a/src/components/Navigation.ts
+++ b/src/components/Navigation.ts
@@ -13,7 +13,7 @@ const Navigation = (props: any, { slots, attrs }: any) => {
   const minSlide = inject('minSlide', ref(1))
   const normalizeDir = inject('normalizeDir', ref('ltr'))
   const currentSlide = inject('currentSlide', ref(1))
-  const nav: CarouselNav = inject('nav', {})
+  const nav: CarouselNav = inject('nav', {} as CarouselNav)
 
   const { wrapAround, i18n } = config
   const getPrevIcon = (): string => {

--- a/src/components/Pagination.ts
+++ b/src/components/Pagination.ts
@@ -11,7 +11,7 @@ const Pagination = () => {
   const maxSlide = inject('maxSlide', ref(1))
   const minSlide = inject('minSlide', ref(1))
   const currentSlide = inject('currentSlide', ref(1))
-  const nav: CarouselNav = inject('nav', {})
+  const nav: CarouselNav = inject('nav', {} as CarouselNav)
 
   const isActive = (slide: number): boolean =>
     mapNumberToRange({

--- a/src/index.ts
+++ b/src/index.ts
@@ -5,5 +5,5 @@ import Navigation from './components/Navigation'
 import Pagination from './components/Pagination'
 import Slide from './components/Slide'
 
-export * from '@/types'
+export type * from '@/types'
 export { Carousel, Slide, Navigation, Pagination, Icon }

--- a/src/types/carousel.ts
+++ b/src/types/carousel.ts
@@ -36,29 +36,29 @@ export interface CarouselConfig {
   i18n: { [key in I18nKeys]?: string }
 }
 
-// Carousel navigation methods
 export interface CarouselNav {
   slideTo: (index: number) => void
   next: () => void
   prev: () => void
 }
 
-export interface CarouselExposed {
+export interface CarouselData {
+  config: CarouselConfig
+  slidesCount: Ref<number>
+  slideSize: Ref<number>
+  currentSlide: Ref<number>
+  maxSlide: Ref<number>
+  minSlide: Ref<number>
+  middleSlide: Ref<number>
+}
+
+export interface CarouselMethods extends CarouselNav {
   updateBreakpointsConfig: () => void
   updateSlidesData: () => void
   updateSlideSize: () => void
   restartCarousel: () => void
-  slideTo: (index: number) => void
-  next: () => void
-  prev: () => void
+}
+export interface CarouselExposed extends CarouselMethods {
   nav: CarouselNav
-  data: {
-    config: CarouselConfig
-    slidesCount: Ref<number>
-    slideSize: Ref<number>
-    currentSlide: Ref<number>
-    maxSlide: Ref<number>
-    minSlide: Ref<number>
-    middleSlide: Ref<number>
-  }
+  data: CarouselData
 }

--- a/src/types/carousel.ts
+++ b/src/types/carousel.ts
@@ -1,3 +1,5 @@
+import { Ref } from 'vue'
+
 import {
   BREAKPOINT_MODE_OPTIONS,
   DIR_OPTIONS,
@@ -34,10 +36,29 @@ export interface CarouselConfig {
   i18n: { [key in I18nKeys]?: string }
 }
 
+// Carousel navigation methods
 export interface CarouselNav {
-  [key: string]: any
+  slideTo: (index: number) => void
+  next: () => void
+  prev: () => void
 }
 
-export interface ElementStyleObject {
-  [key: string]: any
+export interface CarouselExposed {
+  updateBreakpointsConfig: () => void
+  updateSlidesData: () => void
+  updateSlideSize: () => void
+  restartCarousel: () => void
+  slideTo: (index: number) => void
+  next: () => void
+  prev: () => void
+  nav: CarouselNav
+  data: {
+    config: CarouselConfig
+    slidesCount: Ref<number>
+    slideSize: Ref<number>
+    currentSlide: Ref<number>
+    maxSlide: Ref<number>
+    minSlide: Ref<number>
+    middleSlide: Ref<number>
+  }
 }

